lib/global-styles.php

```diff
@@ -488,18 +542,18 @@ function gutenberg_experimental_global_styles_enqueue_assets() {
  * @return array New block editor settings
  */
 function gutenberg_experimental_global_styles_settings( $settings ) {
-	if ( ! gutenberg_experimental_global_styles_has_theme_json_support() ) {
-		return $settings;
-	}
-
 	$settings['__experimentalGlobalStylesUserEntityId'] = gutenberg_experimental_global_styles_get_user_cpt_id();
 
-	$global_styles = gutenberg_experimental_global_styles_merge_trees(
+	$global_styles_base = gutenberg_experimental_global_styles_merge_trees(
 		gutenberg_experimental_global_styles_get_core(),
 		gutenberg_experimental_global_styles_get_theme()
 	);
 
-	$settings['__experimentalGlobalStylesBase'] = $global_styles;
+	$settings['__experimentalGlobalStylesBase'] = $global_styles_base;
+	$settings['__experimentalGlobalStyles']     = gutenberg_experimental_global_styles_merge_trees(
```

Comment:
I'd rather add this in the context when it's necessary.

Reply:
In https://github.com/WordPress/gutenberg/pull/22722 I'm using context as a way to control inherit styles. But still, we need to access to a structure that contains default, theme and user styles merged, this line is just doing that adding this structure. __experimentalGlobalStylesBase does not contain the user styles so if we used the normal editor view would not reflect the user-configured global styles.